src/FilterDetails/FilterDetails.tsx

```diff
@@ -2,50 +2,81 @@ import cn from "classnames";
 import * as React from "react";
 import { neutralZoneClass } from "../HOCs";
 import { IconSvg } from "../IconSvg";
-import { CallbackProps, FilterDetailsProps, State } from "./types";
-
+import { setLoopFocusElements } from "../utils/Focus";
 import "./FilterDetails.scss";
+import { CallbackProps, FilterDetailsProps } from "./types";
 
 type Props = FilterDetailsProps & CallbackProps;
 
 const ESC_KEY = 27;
 
-export class FilterDetails extends React.Component<Props, State> {
-	public state = {
-		helpIsExpanded: false
-	};
+export const FilterDetails: React.FC<Props> = ({
+	onClose,
+	editorComponent,
+	helpComponent,
+	helpCaption,
+	viewMode
+}) => {
+	const [helpIsExpanded, setHelpIsExpanded] = React.useState(false);
+	const kitFiltrationHelperRef = React.useRef<HTMLDivElement>(null);
+	const kitFiltrationExtendButton = React.useRef<HTMLButtonElement>(null);
+	const kitFiltrationRef = React.useRef<HTMLDivElement>(null);
+	const kitEditorWrapperRef = React.useRef<HTMLDivElement>(null);
+
+	React.useEffect(() => {
+		document.addEventListener("keydown", handleKeyDown);
+
+		if (kitEditorWrapperRef.current) {
+			kitEditorWrapperRef.current.addEventListener(
+				"focusin",
+				setLoopFocusElements(kitEditorWrapperRef.current)
+			);
+			kitEditorWrapperRef.current.addEventListener(
+				"keydown",
+				setLoopFocusElements(kitEditorWrapperRef.current)
+			);
+		}
 
-	private kitFiltrationHelperRef = React.createRef<HTMLDivElement>();
-	private kitFiltrationExtendButton = React.createRef<HTMLButtonElement>();
-	private kitFiltrationRef = React.createRef<HTMLDivElement>();
+		return () => {
+			document.removeEventListener("keydown", handleKeyDown);
 
-	public componentDidMount() {
-		document.addEventListener("keydown", this.handleKeyDown);
-		this.handleViewExtended();
-	}
+			if (kitEditorWrapperRef.current) {
+				kitEditorWrapperRef.current.removeEventListener(
+					"focusin",
+					setLoopFocusElements(kitEditorWrapperRef.current)
+				);
+				kitEditorWrapperRef.current.removeEventListener(
+					"keydown",
+					setLoopFocusElements(kitEditorWrapperRef.current)
+				);
+			}
+		};
+	});
 
-	public componentWillUnmount() {
-		document.removeEventListener("keydown", this.handleKeyDown);
-	}
+	React.useEffect(
+		() => {
+			handleViewExtended();
+		},
+		[helpIsExpanded]
+	);
 
-	public handleKeyDown = (e: KeyboardEvent) => {
-		const { onClose } = this.props;
+	const handleKeyDown = (e: KeyboardEvent) => {
 		if (
-			document.activeElement === this.kitFiltrationRef.current &&
+			document.activeElement === kitFiltrationRef.current &&
 			e.keyCode === ESC_KEY
 		) {
 			e.preventDefault();
 			onClose();
 		}
 	};
 
-	public handleHelpExtended = () => {
-		this.setState(state => ({ helpIsExpanded: !state.helpIsExpanded }));
+	const handleHelpExtended = () => {
+		setHelpIsExpanded(newState => !newState);
```

Comment:
Тока тут `prevState`
```suggestion
		setHelpIsExpanded(prevState => !prevState);
```